### PR TITLE
Initial Dev Onboarding + Resources

### PR DESCRIPTION
**Pull request by shubsengupta**
This will serve as a living doc for dev onboarding resources + best practices for contributors to Pathways To Education repositories.

### CODE REVIEW

**Review by orangecubedigital — CHANGES_REQUESTED** [review]
Would suggest linking ProductBoard

**Comment by shubsengupta** [comment]
> Would suggest linking ProductBoard

Good call, on it. Thanks!

File: README.md [hunk 1]
```diff
@@ -1 +1,37 @@
-# get-started
\ No newline at end of file
+# Welcome to the Pathways to Education Lab
+We create + share technologies optimized for impact in the areas of youth well-being and education.
```
Comment (jasonshim) [inline README.md:2]:
```suggestion
We create + share technology optimized for impact in the areas of youth well-being and education. 
```

File: README.md [hunk 2]
```diff
@@ -1 +1,37 @@
...
+# Welcome to the Pathways to Education Lab
+We create + share technologies optimized for impact in the areas of youth well-being and education. 
+
+Thank you for contributing to our product development team - we’re thrilled to have you on board as we build the future of education. Here are some notes and best practices to keep us in sync on this journey.
+
+## Development Best Practices
+-  New to Github? [Get started here](https://guides.github.com/activities/hello-world/)
+- Please share all code in repositories under the [`pathwaystoeducation`](https://github.com/pathwaystoeducation) Github, and use branches to work independently. While not necessary, we recommend naming branches appropriately and [kebab-casing](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841) names (i.e. `gradball-frontend`)
+- Commit early, and commit often; provide short, insightful commit messages that note what's being added/modified and why
+  (e.g. "**Add UI for Login Page**", or "**Remove Extra Character to Fix #21**", referencing Github Issues if needed)
```
Comment (jasonshim) [inline README.md:10]:
```suggestion
  (e.g. "**Add UI for Student Login Page**", or "**Remove Extra Character to Fix #21**", referencing Github Issues if needed) 
```
Including addition of word student to add context b/c this is particular to Pathways. :)

File: README.md [hunk 3]
```diff
@@ -1 +1,37 @@
...
+
+## Development Best Practices
+-  New to Github? [Get started here](https://guides.github.com/activities/hello-world/)
+- Please share all code in repositories under the [`pathwaystoeducation`](https://github.com/pathwaystoeducation) Github, and use branches to work independently. While not necessary, we recommend naming branches appropriately and [kebab-casing](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841) names (i.e. `gradball-frontend`)
+- Commit early, and commit often; provide short, insightful commit messages that note what's being added/modified and why
+  (e.g. "**Add UI for Login Page**", or "**Remove Extra Character to Fix #21**", referencing Github Issues if needed) 
+- Always add as much documentation as possible - with comments in code, in readmes, wikis, etc.
+- Create pull-requests and always ensure PR (Pull Request) is reviewed and approved before merging. This should at the very minimum be you reviewing your own code as a third party, but ideally involves at least one other member of the team
```
Comment (jasonshim) [inline README.md:12]:
```suggestion
- Create pull-requests and always ensure PR (Pull Request) is reviewed and approved before merging. At the very minimum, this should be a thorough review of your own code, but ideally involves a third-party review by at least one other member of the team.
```

File: README.md [hunk 4]
```diff
@@ -1 +1,37 @@
...
+
+### Sample Development Workflow
+
+**Before development:**
+- Feedback is sent in email, slack, or other medium and should be logged as an Insight in [ProductBoard](https://pathwayscanada.productboard.com/feature-board/1972714-feature-organization) *(Requires Login)*
```
Comment (jasonshim) [inline README.md:17]:
```suggestion
- Feedback sent in email, Slack, or other channels and should be logged as an Insight in [ProductBoard](https://pathwayscanada.productboard.com/feature-board/1972714-feature-organization) *(Requires Login)*
```

File: README.md [hunk 5]
```diff
@@ -1 +1,37 @@
...
+- Insights are collected and summarized weekly, and then logged as a Feature in ProductBoard. The colour next to the feature indicates current status
+- When ready to be worked on, features are pushed as Issues to their respective Github repository. As issues are being worked on, the status will be changed in ProductBoard.
+
+**During development:**
+- When working on an issue, please assign the Github Issue to yourself. Share comments or updates within the issue itself as necessary to keep others informed
+- Code should be worked on in a separate branch based on the latest master, unless absolutely required otherwise
```
Comment (jasonshim) [inline README.md:23]:
```suggestion
- Code should be worked on in a separate branch based on the latest main, unless absolutely required otherwise
```

Comment (jasonshim) [reply README.md:23]: To reflect Github change https://github.com/github/renaming

Comment (shubsengupta) [reply README.md]: Good catch, thank you!

File: README.md [hunk 6]
```diff
@@ -1 +1,37 @@
...
+- Please name Branches and Pull Requests as per the guidelines above in Development Best Practices
+- Once the bug-fix is completed, please push the branch, rebasing off master if needed, and then create a pull-request with some brief information and referencing the Issue # in the PR Title
+- Wait for review, tagging other developers if necessary
+- Address any feedback, and merge PRs only when approved by at least one reviewer by using "Squash and Merge."
```
Comment (jasonshim) [inline README.md:27]: Squash and Merge - may need more context? Is there a link or info that can be provided? Or is it assumed that anyone reading would inherently know what it references?

Comment (shubsengupta) [reply README.md:27]: Good call - it's a dropdown when merging a PR, but I can link to it for now. For future, we can enable this as a default for branches so we'll just use that!

**Review by jasonshim — APPROVED** [review]
Looks good!

**Pull request merged**